src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,215 @@
+        Node node = first;
+        for (int i = 0; i < size; i++) {
+            if (index == i) {
+                replacedElement = (T) node.getItem();
+                node.setItem(value);
+                return replacedElement;
+            }
+            node = node.getNext();
+        }
         return null;
     }
 
     @Override
     public T remove(int index) {
+        validateIndex(index);
+        final T removedElement;
+        Node node = first;
+        for (int i = 0; i < size; i++) {
+            if (index == i) {
+                if (i == 0 && size == 1) {
+                    removedElement = (T) node.getItem();
+                    first = null;
+                } else if (i == 0) {
+                    removedElement = (T) node.getItem();
+                    node.getNext().setPrev(node.getPrev());
+                    first = node.getNext();
+                } else if (i == size - 1) {
+                    removedElement = (T) node.getItem();
+                    node.getPrev().setNext(node.getNext());
+                    last = node.getPrev();
+                } else {
+                    removedElement = (T) node.getItem();
+                    node.getPrev().setNext(node.getNext());
+                    node.getNext().setPrev(node.getPrev());
+                }
+                size--;
+                return removedElement;
+            }
+            node = node.getNext();
+        }
         return null;
     }
 
     @Override
     public boolean remove(T object) {
+        Node node = first;
+        for (int i = 0; i < size; i++) {
+            if (node.item == object || (node.item != null && node.item.equals(object))) {
+                if (i == 0 && size == 1) {
+                    first = null;
+                } else if (i == 0) {
+                    node.getNext().setPrev(node.getPrev());
+                    first = node.getNext();
+                } else if (i == size - 1) {
+                    node.getPrev().setNext(node.getNext());
+                    last = node.getPrev();
+                } else {
+                    node.getPrev().setNext(node.getNext());
+                    node.getNext().setPrev(node.getPrev());
+                }
+                size--;
+                return true;
+            }
+            node = node.getNext();
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void validateIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Incorrect index provided");
+        }
+    }
+
+    private void validateAddIndex(int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Incorrect index provided");
+        }
+    }
+
+    private static class Node<E> {
```

Comment:
In this case use T, change it everywhere
```suggestion
    private static class Node<T> {
```
![image](https://github.com/mate-academy/jv-linked-list/assets/48316721/b91fc45d-b62c-4607-828b-ed54c73725c9)

Reply:
done